cloud.foundry.cli/src/test/java/cloud/foundry/cli/logic/GetLogicTest.java

```diff
@@ -85,14 +82,14 @@ public void testGetAllWithConfigurationData() {
         SpaceDevelopersOperations mockSpaceDevelopers = mockSpaceDevelopersOperations();
         ServicesOperations mockServices = mockServicesOperations();
         ApplicationsOperations mockApplications = mockApplicationOperations();
-        ClientOperations mockClientOperations = mockClientOperations();
+        PropertyUtils mockPropertyUtils = mockPropertyUtils();
```

Comment:
I'm pretty sure both the singleton pattern and a fully static class would work here, too. Using a singleton will be a little more work, though, as you'll not only have to subclass here (with a `protected` constructor in the parent class, you can easily write your own class), you will also have to mock the parent class's getter to return your instance. But it's a cleaner approach. I wouldn't mind either.

Reply:
I think there is no need for a singleton in this case. It just increases the complexity imo.

Reply:
It's a matter of style. There is no point in recreating new instances, and your instance keeps giving the exactly same response. That's what singletons have been invented for.

Reply:
It is not what singletons have been invented for imo.